src/LocationQuestionAdminRegionComponent.tsx

```diff
@@ -0,0 +1,137 @@
+import { JsonQLQuery } from "jsonql"
+import { ExprCompiler, Schema } from "mwater-expressions"
+import MWaterDataSource from "mwater-expressions/lib/MWaterDataSource"
+import React, { useEffect, useState } from "react"
+
+type LocationQuestionAdminRegionComponentProps = {
+  apiUrl?: string
+  login?: any
+
+  location: { longitude: number; latitude: number; method: string }
+}
+
+const LocationQuestionAdminRegionComponent: React.FC<LocationQuestionAdminRegionComponentProps> = ({
+  apiUrl,
+  login,
+  location
+}) => {
+  const [adminRegion, setAdminRegion] = useState<string | null>(null)
+  useEffect(() => {
+    if (apiUrl && login) {
+      const jsonql = generateJsonql(location)
+
+      ;(async () => {
+        const datasource = new MWaterDataSource(apiUrl, login.client)
+        const rows = await datasource.performQuery(jsonql)
```

Comment:
Wait, if this fails, it will cause a crash, no? The app needs to be able to view responses offline.

Reply:
wrapped in try catch